src/conventions/h-group/take-action.js

```diff
@@ -130,6 +130,44 @@ export function find_all_clues(game) {
 	];
 }
 
+/**
+ * Returns the list of players who could give the given clue.
+ * @param {Game} game
+ * @param {Clue} clue
+ * @returns {number[]}
+ */
+export function find_clue_givers(game, clue) {
+	const { state } = game;
+	const { result } = clue;
+
+	const givers = [state.ourPlayerIndex];
+	for (let playerIndex = state.nextPlayerIndex(state.ourPlayerIndex);
```

Comment:
If we want this function to generalize, should we start at player after the clue giver (not always us)?

Reply:
The Clue type which we get from `find_clues` doesn't include the giver, but I've added giver as a parameter.